proxy.go

```diff
@@ -191,7 +230,26 @@ func NewProxyHttpServer() *ProxyHttpServer {
 		}),
 		Tr: &http.Transport{TLSClientConfig: tlsClientSkipVerify, Proxy: http.ProxyFromEnvironment},
 	}
-	proxy.ConnectDial = dialerFromEnv(&proxy)
+
+	cfg := httpproxy.FromEnvironment()
```

Comment:
I'm not sure about silently overriding environment proxy variables. WDYT?

Reply:
Also could we name this something to signal what it actually is instead of just `cfg`?